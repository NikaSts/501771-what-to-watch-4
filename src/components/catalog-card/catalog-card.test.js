import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import history from '../../history';
import CatalogCard from './catalog-card';

const title = `Harry Potter`;
const children = <video />;
const id = 1;

it(`MovieCard should render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <CatalogCard
            id={id}
            title={title}
            onMouseEnter={() => {}}
            onMouseLeave={() => { }}
          >
            {children}
          </CatalogCard>
        </Router>
    )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
