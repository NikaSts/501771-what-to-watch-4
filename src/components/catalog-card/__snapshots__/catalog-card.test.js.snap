// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Harry Potter
    </a>
  </h3>
</article>
`;
